extensions/dataspace-authority-spi/src/main/java/org/eclipse/dataspaceconnector/registration/authority/spi/ParticipantVerifier.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.registration.authority.spi;
+
+import org.eclipse.dataspaceconnector.spi.response.StatusResult;
+
+/**
+ * Verifier that fetches and evaluates verifiable credentials to
+ * authorize dataspace membership.
+ */
+@FunctionalInterface
+public interface ParticipantVerifier {
+    /**
+     * Apply dataspace membership policy to evaluate whether a participant may be onboarded.
+     * <p>
+     * The method returns a successful {@link StatusResult} if the policy could be evaluated.
+     * The payload then contains the policy evaluation result.
+     * A failure result indicates that the policy could not be evaluated,
+     * for example, because Verifiable Credentials could not be retrieved from the Identity Hub.
+     *
+     * @param did The putative participant's DID URL. Used to resolve participant Identity Hub and retrieve Verifiable Credentials.
+     * @return a result indicating whether the the policy could be evaluated, and if successful, the evaluation result.
```

Comment:
IMO this is not how the `Result` class was intended. In case the evaluation is not successful (for whatever reason), the `Result` should carry the error in its failure messages property. Thus, returning `Result.success()` indicates a successful evaluation and the method return type should be `Result<Void>`. Or am I overlooking something?

Reply:
The point is that the method does I/O which could fail, and then applies a policy which says true or false.

So the Result is used to indicate whether the method was actually able to compute a result, while the payload tells if the policy allowed or not. This is used to end up in state DENIED vs FAILED

Reply:
NB this implements the state machine from https://github.com/eclipse-dataspaceconnector/RegistrationService/tree/main/docs/developer/decision-records/2022-08-02-participant-status

Reply:
I got that, my concern is only that this slightly differs from how we typically use the `Result`, and could therefor lead users to believe that `Result.success()` indicates that the policy got evaluated to `true`.

At the very least I would put some javadoc on `ParticipantVerifier#isOnboardingAllowed`

Reply:
@algattik is it possible that the Javadoc on `ParticipantVerifier#isOnboardingAllowed` was present previously and somehow got deleted? Or is my git messed up?

Reply:
@paullatzelsperger thank you, this was a poor merge from my part, I've added and expanded the Javadoc